controllers/datadogagent/feature/clusterchecks/feature.go

```diff
@@ -0,0 +1,150 @@
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package clusterchecks
+
+import (
+	apicommon "github.com/DataDog/datadog-operator/apis/datadoghq/common"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/common/v1"
+	datadoghqv1alpha1 "github.com/DataDog/datadog-operator/apis/datadoghq/v1alpha1"
+	"github.com/DataDog/datadog-operator/apis/datadoghq/v2alpha1"
+	apiutils "github.com/DataDog/datadog-operator/apis/utils"
+	"github.com/DataDog/datadog-operator/controllers/datadogagent/feature"
+	corev1 "k8s.io/api/core/v1"
+)
+
+func init() {
+	err := feature.Register(feature.ClusterChecksIDType, buildClusterChecksFeature)
+	if err != nil {
+		panic(err)
+	}
+}
+
+type clusterChecksFeature struct {
+	enable                 bool
+	useClusterCheckRunners bool
+}
+
+func buildClusterChecksFeature(options *feature.Options) feature.Feature {
+	return &clusterChecksFeature{}
+}
+
+func (f *clusterChecksFeature) Configure(dda *v2alpha1.DatadogAgent) feature.RequiredComponents {
+	if dda != nil && dda.Spec.Features != nil && dda.Spec.Features.ClusterChecks != nil {
+		f.enable = apiutils.BoolValue(dda.Spec.Features.ClusterChecks.Enabled)
+		f.useClusterCheckRunners = f.enable && apiutils.BoolValue(dda.Spec.Features.ClusterChecks.UseClusterChecksRunners)
+	}
+
+	if f.enable {
+		return feature.RequiredComponents{
+			ClusterAgent:        feature.RequiredComponent{IsRequired: apiutils.NewBoolPointer(true)},
+			ClusterChecksRunner: feature.RequiredComponent{IsRequired: &f.useClusterCheckRunners},
+		}
+	}
+
+	// Don't set ClusterAgent here because we can have a DCA deployed (as
+	// defined in the "default" feature) with cluster checks disabled.
+	return feature.RequiredComponents{
+		ClusterChecksRunner: feature.RequiredComponent{IsRequired: apiutils.NewBoolPointer(false)},
+	}
+}
+
+func (f *clusterChecksFeature) ConfigureV1(dda *datadoghqv1alpha1.DatadogAgent) feature.RequiredComponents {
+	if dda != nil && dda.Spec.ClusterAgent.Config != nil {
+		f.enable = apiutils.BoolValue(dda.Spec.ClusterAgent.Config.ClusterChecksEnabled)
+		f.useClusterCheckRunners = f.enable && apiutils.BoolValue(dda.Spec.ClusterChecksRunner.Enabled)
+	}
+
+	if f.enable {
+		return feature.RequiredComponents{
+			ClusterAgent:        feature.RequiredComponent{IsRequired: apiutils.NewBoolPointer(true)},
+			ClusterChecksRunner: feature.RequiredComponent{IsRequired: &f.useClusterCheckRunners},
+		}
+	}
+
+	return feature.RequiredComponents{
+		ClusterChecksRunner: feature.RequiredComponent{IsRequired: apiutils.NewBoolPointer(false)},
+	}
+}
+
+func (f *clusterChecksFeature) ManageDependencies(managers feature.ResourceManagers, components feature.RequiredComponents) error {
+	return nil
+}
+
+func (f *clusterChecksFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	if f.enable {
```

Comment:
I believe this is only called if the feature is enabled so we don't need to check for `f.enable`

Reply:
You're right. I've deleted that and simplified the code a bit 👍